src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -1,631 +1,220 @@
-        timeTaken = null;
-    }
-
-    private String changeRoleToAutoIfAutomatedActions(String servletName, String role) {
-        return servletName.startsWith("/auto/") ? "Auto" : role;
-    }
+    private String logMessage;
     
-    private String formatTimeForId(Date date) {
-        SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMddHHmmssSS");
-        sdf.setTimeZone(TimeZone.getTimeZone(Const.SystemParams.ADMIN_TIME_ZONE));
-        return sdf.format(date.getTime());
-    }
-
-    public String getIconRoleForShow() {
-        StringBuilder iconRole = new StringBuilder(100);
-        
-        if (role.contains("Instructor")) {
-            iconRole.append("<span class = \"glyphicon glyphicon-user\" style=\"color:#39b3d7;\"></span>");
-            if (role.contains("(M)")) {
-                iconRole.append("-<span class = \"glyphicon glyphicon-eye-open\" style=\"color:#E61E1E;\"></span>- ");
-            }
-        } else if (role.contains("Student")) {
-            iconRole.append("<span class = \"glyphicon glyphicon-user\" style=\"color:#FFBB13;\"></span>");
-            if (role.contains("(M)")) {
-                iconRole.append("-<span class = \"glyphicon glyphicon-eye-open\" style=\"color:#E61E1E;\"></span>- ");
-            }
-        } else if (role.contains("Unregistered")) {
-            iconRole.append("<span class = \"glyphicon glyphicon-user\"></span>");
-        } else if (role.contains("Auto")) {
-            iconRole.append("<span class = \"glyphicon glyphicon-cog\"></span>");
-        } else {
-            iconRole.append(role);
-        }
-
-        if (role.contains("Admin")) {
-            iconRole.append("<span class = \"glyphicon glyphicon-user\" style=\"color:#E61E1E;\"></span>");
-        }
-
-        return iconRole.toString();
-    }
+    private String actionUrl;
     
-    /**
-     * Assumption: the {@code requestUrl} is in the format "/something/actionName"
-     *   possibly followed by "?something" e.g., "/page/studentHome?user=abc"
-     * @return action name in the URL e.g., "studentHome" in the above example.
-     */
-    public static String getActionName(String requestUrl) {
-        return requestUrl.split("/")[2].split("\\?")[0];
+    private long actionTimeTaken;
+    
+    // legacy of messing up UI and logic
+    private boolean logToShow = true;
+    
+    public ActivityLogEntry(Builder builder) {
+        logTime = builder.logTime;
+        actionServletName = builder.actionServletName;
+        actionTimeTaken = builder.actionTimeTaken;
+        actionName = builder.actionName;
+        userRole = builder.userRole;
+        userName = builder.userName;
+        userGoogleId = builder.userGoogleId;
+        userEmail = builder.userEmail;
+        logMessage = builder.logMessage;
+        actionUrl = builder.actionUrl;
+        logId = builder.logId;
+        actionTimeTaken = builder.actionTimeTaken;
+        isMasqueradeUserRole = builder.isMasqueradeUserRole;
```

Comment:
isMasqueradeUserRole is primitive type, which will be initialised to false by default in JVM.